providers/amazon/src/airflow/providers/amazon/aws/operators/s3.py

```
@@ -366,6 +366,159 @@ def get_openlineage_facets_on_start(self):
+        dest_bucket_name, dest_bucket_prefix = self.hook.get_s3_bucket_key(
+            self.dest_bucket_name, self.dest_bucket_prefix, "dest_bucket_name", "dest_bucket_prefix"
+        )
+
+        # Get paginator
+        s3_client = self.hook.get_conn()
+
+        paginator = s3_client.get_paginator("list_objects_v2")
+        pages = paginator.paginate(
+            Bucket=source_bucket_name,
+            Prefix=source_bucket_prefix,
+            PaginationConfig={"PageSize": self.page_size},
+        )
+
+        # Copy objects
+        copied_object_count = 0
+        failed_object_count = 0
+        for page in pages:
+            if "Contents" in page:
+                for obj in page["Contents"]:
+                    source_key = obj["Key"]
+                    dest_key = source_key.replace(source_bucket_prefix, dest_bucket_prefix, 1)
+
+                    try:
+                        self.hook.copy_object(
+                            source_bucket_key=source_key,
+                            dest_bucket_key=dest_key,
+                            source_bucket_name=source_bucket_name,
+                            dest_bucket_name=dest_bucket_name,
+                            acl_policy=self.acl_policy,
+                            meta_data_directive=self.meta_data_directive,
+                        )
+
+                        copied_object_count += 1
+                    except Exception as e:
+                        if self.continue_on_failure:
+                            self.log.error("Failed to copy %s: %s", source_key, e)
+                            failed_object_count += 1
+                        else:
+                            raise RuntimeError(f"Failed to copy {source_key}: {e}")
+
+        self.log.info("Successfully copied %s object(s)", copied_object_count)
+
+        if failed_object_count > 0:
+            raise RuntimeError(f"Failed to copy {failed_object_count} object(s)")
```

My expectation would be that `continue_on_failure` would not fail the task, perhaps just log at ERROR or WARNING that some copy operations failed. But I also see the exception approach as being a "you can't miss this" communication mechanism that some things failed.

Curious what others thing CC @vincbeck @ferruzzi

---

Personally, I'm not a big fan of silent failures, so my direction with `continue_on_failure` was more "do I want to copy as much data as possible in the event of a failure or just stop immediately?". In both cases, there has been an error so the operator should fail, it's just a matter of when. I think there are valid scenarios in which incomplete data is better than no data, but one still wants to be alerted about issues.

Of course I do understand your interpretation and I think it also makes sense. Would be great to hear other opinions as well @vincbeck @ferruzzi .